django/contrib/auth/views.py

```diff
@@ -122,9 +124,17 @@ def dispatch(self, request, *args, **kwargs):
             return HttpResponseRedirect(next_page)
         return super().dispatch(request, *args, **kwargs)
 
+    def get(self, request, *args, **kwargs):
```

Comment:
@felixxm / @carltongibson  I think we should:
 * Move this warning into `dispatch` inside an `if request` check
 * Move the actual code from `dispatch` into post
 * set `get = post`
 * and alter `http_method_names` to `['get', 'head', 'post', 'options']`
 
 After the deprecation period we would:
 * Drop the warning
 * Drop the `get = post`
 * Remove `get & head` from the `http_method_names`.
 
WDYT?

Reply:
We need to remove `get` from `http_method_names` because `TemplateView` still defines it and I really don't want it to be callable :) Or We stop using `TemplateView` and inherit from `ContextMixin` ` TemplateResponseMixin ` and `View` ourself…

Reply:
Implemented.